SourceS/sdl_compat.h

```diff
@@ -94,6 +80,12 @@ inline int SDLC_SetSurfaceAndPaletteColors(SDL_Surface *surface, SDL_Palette *pa
 	}
 	if (colors != (palette->colors + firstcolor))
 		SDL_memcpy(palette->colors + firstcolor, colors, ncolors * sizeof(*colors));
+
+	#if SDL1_VIDEO_MODE_BPP == 8
+		// In SDL1, if the Video Surface is 8bit, you need to set this as well to match,
+		// otherwise colors will be wrong!
+		SDL_SetColors(SDL_GetVideoSurface(), colors, firstcolor, ncolors);

```

Comment:
Is this different from `SDL_SetPalette(surface, SDL_LOGPAL | SDL_PHYSPAL, ...`?
https://www.libsdl.org/release/SDL-1.2.15/docs/html/sdlsetcolors.html

Reply:
Don't really know. This is one of the first change I made upon Arczi's advice in the beginning to get the colors right. From my inspecting SDL sources, it seem that the surface palette is'nt automatically copied onto the VideoSurface (missing a version counter in palette maybe?), and then SDL does approximates the surface colors.

Let me test with the SDL_PHYSPAL as well...

Nope: it doesn't work with
![image](https://user-images.githubusercontent.com/33961444/71269156-edaa3e80-234e-11ea-8cf1-9f44023cd1bb.png)
![image](https://user-images.githubusercontent.com/33961444/71269182-fdc21e00-234e-11ea-8149-f56bbe511bcb.png)

Reply:
Ok, in that case we can upstream that commit directly. Can you also take a look at SDLC_SetSurfaceColors since it has also been made to use SDL_PHYSPAL.

Reply:
Thanks for going over this, the fix has not been cherry-picked upstream:
https://github.com/diasurgical/devilutionX/commit/319bdeddd242bb79c5145fd81ee68d584d4b220e